### PR TITLE
Prevent gas oveflow in the memory expansion cost computation

### PR DESCRIPTION
**Pull request by LuisPH3**
Use the same method as [Geth](https://github.com/Fantom-foundation/tosca-go-ethereum/blob/4db0f269681aa0137a5510532a8c4f0c423237e7/core/vm/gas_table.go#L39):

- Verify that overflow can still not happen, although LFVM gas is signed. 
- This is related to #524, may need to be reverted to investigate the cause of the CT not detecting this bug.

### CODE REVIEW

File: go/vm/lfvm/memory.go [hunk 1]
```diff
@@ -44,14 +44,24 @@ func toValidMemorySize(size uint64) uint64 {
 	return fullWordsSize
 }
 
+const (
+	// Maximum memory size allowed
+	// This magic number comes from 'core/vm/gas_table.go' 'memoryGasCost' in geth
```
Comment (facuMH) [inline go/vm/lfvm/memory.go:49]: thank you very much for this comment.

File: go/vm/lfvm/memory_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	"math"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/vm"
+)
+
+func TestMagicNumber(t *testing.T) {
+
+	// Call me paranoid, but I am repeating the test for the magic number:
+	// The difference between geth and lfvm is that they use uint64 as gas
```
Comment (facuMH) [inline go/vm/lfvm/memory_test.go:23]:
```suggestion
	// The difference between geth and lfvm is that geth uses uint64 as gas
```

**Review by facuMH — APPROVED** [review]
LGTM

**Review by facuMH — APPROVED** [review]

File: go/vm/lfvm/memory.go [hunk 3]
```diff
@@ -44,14 +44,33 @@ func toValidMemorySize(size uint64) uint64 {
...
```
Comment (HerbertJordan) [inline go/vm/lfvm/memory.go:49]: You could de-magicfy it a bit by stating that this is the largest amount of memory that can be allocated such that the expansion costs do not exceed `MaxInt64`.

**Review by HerbertJordan — APPROVED** [review]
Thanks for fixing this!

**Review by simonlechner — APPROVED** [review]

**Pull request merged**